scripts/session.js

```diff
@@ -1,165 +1,186 @@
-  // data: 'where=' + JSON.stringify({"name":sessionName}),
-  // success: function(data){
-    // $("#session_name").text(sessionName);
-    // var points = data.results[0].points;
-    // for(var i = 0 ; i < points.length; i++) {
-      // var row$ = $('<button type="button" class="btn btn-info points">');
-      // row$.html(points[i].label);
-      // $("#point_labels").append(row$);
-    // }
-  // }
-// });
-
-// show points label list
-$("#session_name").text(sessionName);
-for(var i = 0 ; i < defaultPoints.length; i++) {
-  var row$ = $('<button type="button" class="btn btn-info points">');
-  row$.html(defaultPoints[i].label);
-  $("#point_labels").append(row$);
-}
+    // redirect to login page
+    if (!this.sessionName) {
+      location.href = '/index.html';
+    }
 
-// append user function
-var append_user = function(user_name, point) {
-  if($('td[id=' + user_name + ']')[0] == undefined){
-    if(point == undefined || point == "undefined"){
-      $("#user-point-list").append('<tr><td>' + user_name + '</td><td id=' + user_name +' class="hidden-point"></td></tr>');
-    }else{
-      $("#user-point-list").append('<tr><td>' + user_name + '</td><td id=' + user_name +' class="ready-point">' + point + '</td></tr>');
+    if (this.sessionName && !this.userName) {
+      location.href = '/join_session.html';
     }
   }
-}
 
-// show user on page
-// append_user(userName);
-
-// choose a points
-$('#point_labels').on('click', 'button', function(){
-  var point = this.textContent.replace(/[^(\d\.)|?]*/g, '');
-  $('#' + userName).text(point);
-  $('#' + userName).attr('class', 'ready-point');
-  Cookies.set('point', point);
-  ws.send(JSON.stringify({"bc": sessionName, "type":"user_choose_point", "user_name": userName, "point": point}));
-});
-
-// clear all votes
-$('#clear-votes').click(function(){
-  $("td[id]").each(function(){
-    this.innerText = '';
-    this.className = 'hidden-point';
-  });
-  Cookies.remove('point');
-  ws.send(JSON.stringify({"bc": sessionName, "type":"clear_all_votes"}));
-});
-
-// show votes
-$('#show-votes').click(function(){
-  // $('td.hidden-point').removeClass('hidden-point');
-  $("td[id]").each(function(){
-    this.className = '';
-  });
-  ws.send(JSON.stringify({"bc": sessionName, "type":"show_all_votes"}));
-});
-
-// --- WEBSOCKET ---
-//***************************
-//
-// add event handler for incomming message
-ws.onmessage = function(evt){
-  var my_received_message = JSON.parse(evt.data);
-
-  // add new user also sync me to him
-  if(my_received_message.type == "new_connection"){
-    append_user(my_received_message.user_name, my_received_message.point);
-    if(userName != my_received_message.user_name){
-      ws.send(JSON.stringify({"to": my_received_message.user_name, "type": "new_connection_sync", "user_name": userName, "point": cookie_point}));
-    }
+  _initConncetion() {
+    let ws = new WebSocket('ws://achex.ca:4010');
+    let that = this;
```

Comment:
All `that` have been removed in #14 .

The reason I added `that` is because when ES6 was introduced, I used `class`, and I saved some common values with `this` (it's current class).

However, in our code, we always used `this` in jQuery selector to indicate current element. That said, there are conflicts between two `this`s. Hence `that` I added.

In #14 , I found for the latter case, we can use `event.target` in `$('#point-labels').on('click', 'button', (event) => {}` and `element` in `$('td[id]').each((index, element) => {}` to indicate current element instead of using `this`.

In this case, we can safely change all the `function()` to arrow function.